### PR TITLE
Generalize the ColSpec interface

### PR DESCRIPTION
**Pull request by eliascarv**
Generalize the ColSpec type to be used in the transformations: Select, Reject, DropMissing and in future transformations.

### CODE REVIEW

File: test/transforms.jl [hunk 1]
```diff
@@ -107,8 +123,7 @@
     tₒ = revert(T, n, c)
     @test t == tₒ
 
-    # throws: Select without arguments
-    @test_throws ArgumentError Select()
```
Comment (juliohm) [inline test/transforms.jl:111]: What is the current meaning of Select() ?

File: src/transforms/utils.jl [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+const ColSelector = Union{Symbol, Integer, AbstractString}
+const ColSpec = Union{Vector{T}, NTuple{N, T}, Regex, Colon} where {N, T<:ColSelector}
+
+_select(colspec::Vector{Symbol}, allcols) = colspec
+_select(colspec::Vector{<:Integer}, allcols) = allcols[colspec]
+_select(colspec::Vector{<:AbstractString}, allcols) = Symbol.(colspec)
+_select(colspec::NTuple{N, <:ColSelector}, allcols) where {N} =
```
Comment (juliohm) [inline src/transforms/utils.jl:11]:
```suggestion
_select(colspec::NTuple{N,<:ColSelector}, allcols) where {N} =
```

File: src/transforms/utils.jl [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+const ColSelector = Union{Symbol, Integer, AbstractString}
+const ColSpec = Union{Vector{T}, NTuple{N, T}, Regex, Colon} where {N, T<:ColSelector}
```
Comment (juliohm) [inline src/transforms/utils.jl:6]:
```suggestion
const ColSpec = Union{Vector{T}, NTuple{N,T}, Regex, Colon} where {N,T<:ColSelector}
```

**Review by juliohm — COMMENTED** [review]
It would be nice to document a bit further the utils.jl file so that first-time readers can figure out what is happening. Also, I left some comments on specific lines.

File: src/transforms/colspec.jl [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+# because Vector is more flexible than Tuple.
+#
+# note 2: allcols can be of type Vector or Tuple,
+# in some cases it doesn't matter, but in others it does.
+#
+# note 3: all other methods of the function convert their arguments
+# to call the method below (main method), except the last one.
+
+# _filter(::Vector{Symbol}, allcols) - main method
+# second argument can be a Vector or Tuple
```
Comment (juliohm) [inline src/transforms/colspec.jl:23]: I believe that this comment section can be omitted, the code already provides this information.

File: src/transforms/colspec.jl [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+function _filter(colspec::Vector{Symbol}, allcols)
```
Comment (juliohm) [inline src/transforms/colspec.jl:24]: Please rename allcols to cols in all places

**Review by juliohm — CHANGES_REQUESTED** [review]
I believe that this PR is already mixing up too many things. If you can split it up in a PR with the minimum amount of changes to introduce a ColSpec type, we can then write separate PRs updating the Select/Reject, Rename, DropMissing, ...

**Comment by eliascarv** [comment]
I think I'll close this PR and make a new one that just adds the colspec.jl file without changing the Select, Revert and DropMissing.
What do you think?

**Comment by eliascarv** [comment]
This PR will be closed and replaced with smaller PRs.

**Pull request closed without merging**